server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -174,7 +175,7 @@ private void assertSystemMailboxesAreNotUpdated(Mailbox mailbox, MailboxUpdateRe
     @VisibleForTesting
     <T> boolean requestChanged(Optional<T> requestValue, Optional<T> storeValue) {
         return requestValue
-            .filter(value -> !requestValue.equals(storeValue))
```

Comment:
The root cause of the failure ;-)

Reply:
That looks weird right?

Reply:
this is more than weird, because value is not used and requestValue is used instead! Anyway I agree that your change breaks this line

Reply:
We leverage an optional to handle something set to null, something unchaged, or something set to a value. (and it is clearly a bad practice, error prone)